src/psyclone/psyGen.py

```diff
@@ -5509,6 +5619,52 @@ def __init__(self, name, parent=None):
         self._name = name
         self._symbol_table = SymbolTable(self)
 
+    @staticmethod
+    def create(name, symbol_table, children):
+        '''Create a KernelSchedule instance given a name, a symbol table and a
+        list of child nodes.
+
+        :param str name: the name of the KernelSchedule.
+        :param symbol_table: the symbol table associated with this \
+            KernelSchedule.
+        :type symbol_table: :py:class:`psyclone.psyGen.SymbolTable`
+        :param children: a list of PSyIR nodes contained in the \
+            KernelSchedule.
+        :type children: list of :py:class:`psyclone.psyGen.Node`
+
+        :returns: a KernelSchedule instance.
+        :rtype: :py:class:`psyclone.psyGen.KernelInstance`
+
+        '''
+        if not isinstance(name, str):
+            raise GenerationError(
+                "name argument to class KernelSchedule method create "
+                "should be a string but found '{0}'."
+                "".format(type(name).__name__))
+        if not isinstance(symbol_table, SymbolTable):
+            raise GenerationError(
+                "symbol_table argument to class KernelSchedule method create "
+                "should be a SymbolTable but found '{0}'."
+                "".format(type(symbol_table).__name__))
+        if not isinstance(children, list):
+            raise GenerationError(
+                "children argument to class KernelSchedule method "
+                "create should be a list but found '{0}'."
+                "".format(type(children).__name__))
+        for child in children:
+            if not isinstance(child, Node):
+                raise GenerationError(
+                    "child of children argument to class KernelSchedule method "
```

Comment:
pycodestyle: line too long

Reply:
I need to learn to run pycodestyle just before saying ready for review!